### ISSUE TITLE
missing icons in tortoisehg toolbar

### ISSUE THREAD

**Issue by dckc** [NixOS/nixpkgs#13507]
## Basic info

To make sure that we are on the same page:

* Kernel: (run `uname -a`)
Linux pav 3.13.0-79-generic #123-Ubuntu SMP Fri Feb 19 14:27:58 UTC 2016 x86_64 x86_64 x86_64 GNU/Linux

* System: (NixOS: `nixos-version`, Ubuntu/Fedora: `lsb_release -a`, ...)
No LSB modules are available.
Distributor ID:	Ubuntu
Description:	Ubuntu 14.04.4 LTS
Release:	14.04
Codename:	trusty

* Nix version: (run `nix-env --version`)
nix-env (Nix) 1.10

* Nixpkgs version: (run `nix-instantiate --eval '<nixpkgs>' -A lib.nixpkgsVersion`)
"16.03pre76763.be0abb3"

and the app itself is:

/nix/store/gxy7aws40mycs4d1zawxfwbbs5a3430q-tortoisehg-3.6/bin/tortoisehg

## Describe your issue here
Goto Revision and many other icons are missing and fall back to text.

### Expected result
A toolbar full of icons

### Actual result
A toolbar with icons for reload, stop, and home, followed by text buttons for Goto Revision, Filter Toolbar, etc.

secreenshot:

![thg-missing-icons](https://cloud.githubusercontent.com/assets/150986/13374101/a98dc558-dd40-11e5-8e48-c285f0aa385f.png)

### Steps to reproduce
1. `nix-env -i tortoisehg`
2. `tortoisehg`

**Comment by dckc** [comment]
Is there an idiom for "show me the dependency tree of this package"?

**Comment by dckc** [comment]
strace shows:

```
stat("/usr/share/icons/gnome/16x16/actions/go-home.png", {st_mode=S_IFREG|0644, st_size=672, ...}) = 0
...
stat("/usr/share/icons/gnome/16x16/actions/go-to-rev.png", 0x7ffe68d22e30) = -1 ENOENT (No such file or directory)
```

There are lots more similar entries. [thg-trace.txt.gz](https://github.com/NixOS/nixpkgs/files/149561/thg-trace.txt.gz) is attached. To summarize:


```
$ grep icons thg-trace.txt |grep 'ENOENT'|wc
  34736  347360 4185116
$ grep icons thg-trace.txt |grep '= 0'|wc
     74     444    7853
```

**Comment by dckc** [comment]
source path: [icons/scalable/actions/go-to-rev.svg](https://bitbucket.org/tortoisehg/thg/src/b4cbd50340e679eb7602ac931e79fb5f182a34c3/icons/scalable/actions/go-to-rev.svg?at=default&fileviewer=file-view-default)

and it's installed as:
```
$ ls /nix/store/gxy7aws40mycs4d1zawxfwbbs5a3430q-tortoisehg-3.6/share/pixmaps/tortoisehg/icons/scalable/actions/go-to-rev.svg 
/nix/store/gxy7aws40mycs4d1zawxfwbbs5a3430q-tortoisehg-3.6/share/pixmaps/tortoisehg/icons/scalable/actions/go-to-rev.svg
```

but the app isn't looking there:

```
$ grep 'icons' thg-trace.txt |grep 'scalable/actions/go-to-rev'
stat("/usr/share/icons/gnome/scalable/actions/go-to-rev.png", 0x7ffe68d22e30) = -1 ENOENT (No such file or directory)
stat("/usr/share/icons/gnome/scalable/actions/go-to-rev.svg", 0x7ffe68d22e30) = -1 ENOENT (No such file or directory)
stat("/usr/share/icons/hicolor/scalable/actions/go-to-rev.png", 0x7ffe68d22c10) = -1 ENOENT (No such file or directory)
stat("/usr/share/icons/hicolor/scalable/actions/go-to-rev.svg", 0x7ffe68d22c10) = -1 ENOENT (No such file or directory)
stat("/nix/store/d945ibfx9x185xf04b890y4f9g3cbb63-python-2.7.11/share/pixmaps/tortoisehg/icons/scalable/actions/go-to-rev.svg", 0x7ffe68d23270) = -1 ENOENT (No such file or directory)
```

**Comment by dckc** [comment]
I think I see what's going on:

we have
```
     substituteInPlace $sourceRoot/setup.py \
       --replace "/usr/share/" "$out/share/"
```
at [line 20 of tortoisehg/default.nix](https://github.com/NixOS/nixpkgs/blob/master/pkgs/applications/version-management/tortoisehg/default.nix#L20)

but 3.6 was released Dec 9 (at f444d62), and previous to that, on June 8, `setup.py` was changed:

> The installation paths should be calculated from --prefix and --root.

[e8655ee](https://bitbucket.org/tortoisehg/thg/commits/e8655ee7414d4b9df5856b586099e3773f947382
)
